libtransmission/peer-socket.h

```diff
@@ -32,12 +32,32 @@ class tr_peer_socket
     tr_peer_socket() = default;
     tr_peer_socket(tr_session const* session, tr_address const& address, tr_port port, tr_socket_t sock);
     tr_peer_socket(tr_address const& address, tr_port port, struct UTPSocket* const sock);
-    tr_peer_socket(tr_peer_socket&&) = default;
+    tr_peer_socket(tr_peer_socket&& s)
+        : handle{ s.handle }
+        , address_{ s.address_ }
+        , port_{ s.port_ }
+        , type_{ s.type_ }
+    {
+        s.type_ = Type::None;
+        s.handle = {};
+    }
```

Comment:
I like this PR a lot. Just to play code golf a little, what would you think of

```suggestion
    tr_peer_socket(tr_peer_socket&& s)
    {
        *this = std::move(s);
    }
```